kamu-adapter-http/src/ws_smart_transfer_protocol_axum_server.rs

```diff
@@ -0,0 +1,280 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use axum::extract::ws::Message;
+use serde::{de::DeserializeOwned, Serialize};
+use thiserror::Error;
+
+use kamu::domain::{Dataset, BlockRef};
+use url::Url;
+use crate::{
+    messages::*, 
+    ws_common::{WriteMessageError, ReadMessageError, self}, 
+    dataset_protocol_helper::*
+};
+
+
+/////////////////////////////////////////////////////////////////////////////////
+
+
+async fn axum_ws_read_generic_payload<TMessagePayload: DeserializeOwned>(
```

Comment:
for a module-private function this seems like a very verbose name :)

Reply:
Simplified here, both read & write, and similar in Tungstenite part.